src/context/values.js

```diff
@@ -37,35 +38,48 @@ const Pending = {
  * easily available as an array to pass them to the `recursive` and
  * `compute` callbacks without reallocation.
  *
- * @typedef {{
- *   values: !Array,
- *   setters: !Array,
- * }}
+ * @interface
+ * @template T
  */
-let InputDef;
+function InputDef() {}
+/** @type {!Array<T>} */
+InputDef.prototype.values;
+/** @type {!Array<function(T)>} */
+InputDef.prototype.setters;
 
 /**
  * The structure for a property's computed values and subscribers.
- *
- * @typedef {{
- *   prop: !ContextProp,
- *   subscribers: !Array<!Function>,
- *   value: *,
- *   pending: !Pending,
- *   counter: number,
- *   depsValues: !Array,
- *   parentValue: *,
- *   parentContextNode: ?./node.ContextNode,
- *   ping: function(boolean),
- *   pingDep: !Array<function(*)>,
- *   pingParent: ?function(*),
- * }}
+ * @interface
+ * @template T
+ * @template DEP
  */
-let UsedDef;
+function UsedDef() {}
```

Comment:
FMI: did you have to make this format change away from typedef due to incompatibility with templates?

Reply:
Yes